Cue/app/tabs/library/LibraryHome.js

```diff
@@ -156,18 +156,45 @@ class LibraryHome extends React.Component {
     return Object.keys(currentRoute).length === 0
   }
 
+  _shouldThrottleAutoSync = () => {
+    // Throttle calls to _refresh to avoid calling _refresh because:
+    //  1. We can sometimes receive multiple callbacks with [true, false, true]
+    //     in quick succession when the network is reconnecting.
+    //  2. If the SyncConflicts view is dismissed, LibraryHome will re-enter
+    //     the foreground, which can trigger another sync, which will re-push
+    //     SyncConflicts back onto the Navigator stack.
+
+    let msSinceLastSync = this.state.lastSyncTime
+      ? new Date() - this.state.lastSyncTime
+      : Infinity
+    let shouldThrottle = msSinceLastSync < AUTO_SYNC_TRIGGER_THROTTLE_MS
+
+    if (shouldThrottle) {
+      console.warn('_shouldThrottleAutoSync: Auto sync is being throttled (last sync was '
+        + `${msSinceLastSync.toString()} ms ago)`)
+    }
+
+    return shouldThrottle
+  }
+
   _onNavigatorEvent = () => {
     if (this._isLibraryHomeInForeground()) {
       console.info('_onNavigatorEvent: Navigator is at LibraryHome')
 
       // Trigger a sync if needsSync is set or if there are any pending localChanges
       if (this.state.needsSync || this.props.localChanges.length) {
         if (this.state.connected) {
-          console.info('_onNavigatorEvent: Sync is pending and network is available; refreshing now '
-            + `(needsSync: ${this.state.needsSync.toString()}, `
-            + `localChanges.length: ${this.props.localChanges.length.toString()})`)
+          if (!this._shouldThrottleAutoSync()) {
```

Comment:
I wish I could collapse this crazy pyramid but this seems like the easiest way to get good logs